sdk/samples/iot/paho_iot_adu_sample_common.c

```diff
@@ -0,0 +1,662 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/*
+ * Common implementation for *a device* that implements the Model Id
+ * "dtmi:com:example:Thermostat;1".  The model JSON is available in
+ * https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json.
+ *
+ * This code assumes that an MQTT connection to Azure IoT hub and that the underlying
+ * az_iot_hub_client have already been initialized in the variables mqtt_client and hub_client.  The
+ * sample callers paho_iot_pnp_sample.c and paho_iot_pnp_sample_with_provisioning.c do this before
+ * invoking paho_iot_adu_sample_device_implement().
+ *
+ * This should not be confused with ./pnp/pnp_thermostat_component.c.  Both C files implement
+ * The Thermostat Model Id.  In this file, the Thermostat is the only Model that the device
+ * implements.  In ./pnp/pnp_thermostat_component.c, the Thermostat is a subcomponent of a more
+ * complex device and hence the logic is more complex.
+ */
+
+#ifdef _MSC_VER
+// warning C4204: nonstandard extension used: non-constant aggregate initializer
+#pragma warning(disable : 4204)
+// warning C4996: 'localtime': This function or variable may be unsafe.  Consider using localtime_s
+// instead.
+#pragma warning(disable : 4996)
+#endif
+
+#ifdef _MSC_VER
+#pragma warning(push)
+// warning C4201: nonstandard extension used: nameless struct/union
+#pragma warning(disable : 4201)
+#endif
+#include <paho-mqtt/MQTTClient.h>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+#include <stdlib.h>
+#include <time.h>
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+#include <azure/iot/az_iot_adu.h>
+#include <azure/iot/az_iot_hub_client_properties.h>
+
+#include "iot_sample_common.h"
+#include "paho_iot_adu_sample_common.h"
+
+MQTTClient mqtt_client;
+az_iot_hub_client hub_client;
+
+#define DOUBLE_DECIMAL_PLACE_DIGITS 2
+
+#define SAMPLE_MQTT_TOPIC_LENGTH 128
+#define SAMPLE_MQTT_PAYLOAD_LENGTH 1024
+
+bool is_device_operational = true;
+
+// MQTT Connection Values
+static uint16_t connection_request_id = 0;
+static char connection_request_id_buffer[16];
+
+// Telemetry Values
+static az_span const telemetry_name = AZ_SPAN_LITERAL_FROM_STR("count");
+static uint32_t telemetry_count;
+
+// Property Values
+static az_span const property_success_name = AZ_SPAN_LITERAL_FROM_STR("success");
+
+// ADU Feature Values
+static az_iot_adu_update_request xBaseUpdateRequest;
+static az_iot_adu_update_manifest xBaseUpdateManifest;
+static char adu_new_version[16];
+static bool did_parse_update = false;
+static bool did_update = false;
+static char adu_scratch_buffer[10000];
+
+#define AZ_IOT_ADU_AGENT_VERSION "DU;agent/0.8.0-rc1-public-preview"
```

Comment:
Instead of redefining we could just use the homonymous macro from az_iot_adu_client.h